src/service/history-impl.js

```diff
@@ -0,0 +1,659 @@
+/**
```

Comment:
Unfortunately git does not have a way to move a file and keep something in the original place in the same commit.